src/cmdlib.sh

```diff
@@ -392,11 +384,16 @@ runcompose() {
         sudo -E "$@"
         sudo chown -R -h "${USER}":"${USER}" "${tmprepo}"
     else
-        compose_qemu_args=()
-        if [ -f "${workdir}/cache/cache2.qcow2" ]; then
-            compose_qemu_args+=("-drive" "if=none,id=cache,discard=unmap,file=${workdir}/cache/cache2.qcow2" \
-                                "-device" "virtio-blk,drive=cache")
+        # "cache2" has an explicit label so we can find it in qemu easily
+        if [ ! -f "${workdir}"/cache/cache2.qcow2 ]; then
+            flock "${workdir}"/cache/cache2.lock -c "qemu-img create -f qcow2 cache2.qcow2.tmp 10G &&
+               LIBGUESTFS_BACKEND=direct virt-format --filesystem=xfs --label=cosa-cache -a cache2.qcow2.tmp &&
+               mv -T cache2.qcow2.tmp ${workdir}/cache/cache2.qcow2"
         fi
+        # And remove the old one
+        rm -vf cache/cache.qcow2
```

Comment:
Should this include `workdir`?

```suggestion
        rm -vf "${workdir}"/cache/cache.qcow2
```

Reply:
Yes, thank you!